### PR TITLE
Add concurrent adding columns (ALTER TABLE ADD COLUMN) test to BaseConnectorTest

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add concurrent adding columns test to BaseConnectorTest

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #11634

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 1]
```diff
@@ -2847,6 +2847,78 @@ protected void verifyConcurrentInsertFailurePermissible(Exception e)
...
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_add_column", "(col integer)")) {
+            String tableName = table.getName();
+
+            List<Future<OptionalInt>> futures = IntStream.range(0, threads)
+                    .mapToObj(threadNumber -> executor.submit(() -> {
+                        barrier.await(30, SECONDS);
+                        try {
+                            getQueryRunner().execute("ALTER TABLE " + tableName + " ADD COLUMN col" + threadNumber + " integer");
```
Comment (findepi) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:2870]: Declare `String columnName = "col" + threadNumber;`
and make the Future a `Future<Optional<String>>` by retuning the `columnName` on success

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 2]
```diff
@@ -2847,6 +2847,78 @@ protected void verifyConcurrentInsertFailurePermissible(Exception e)
...
+                                    verifyFailure.addSuppressed(e);
+                                }
+                                throw verifyFailure;
+                            }
+                            return OptionalInt.empty();
+                        }
+                    }))
+                    .collect(toImmutableList());
+
+            List<Integer> values = futures.stream()
```
Comment (findepi) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:2889]: `List<Integer> values` -> `List<String> addedColumns`

**Review by findepi — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 3]
```diff
@@ -2847,6 +2847,78 @@ protected void verifyConcurrentInsertFailurePermissible(Exception e)
...
+                    .map(future -> tryGetFutureValue(future, 30, SECONDS).orElseThrow(() -> new RuntimeException("Wait timed out")))
+                    .filter(OptionalInt::isPresent)
+                    .map(OptionalInt::getAsInt)
+                    .collect(toImmutableList());
+
+            if (values.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:2895]: We can require at least one success, but then we may need to provide a connector a way to opt-out.
(i am fine not requiring this, and fine requiring this; and opt-out can be added later, if not needed now)

**Pull request merged**